pandas/tests/dtypes/test_common.py

```diff
@@ -573,6 +574,35 @@ def test_is_extension_type(check_scipy):
         assert not com.is_extension_type(scipy.sparse.bsr_matrix([1, 2, 3]))
 
 
+def test_is_extension_type_deprecation():
+    with tm.assert_produces_warning(FutureWarning):
+        com.is_extension_type([1, 2, 3])
+
+
+@pytest.mark.parametrize(
+    "check_scipy", [False, pytest.param(True, marks=td.skip_if_no_scipy)]
+)
+def test_is_extension_array_dtype(check_scipy):
```

Comment:
I copy pasted the original `is_extension_type` to test `is_extension_array_dtype`, as that is going to replace it and was not yet tested here (so when cleaning this up, can simply remove the above `is_extension_type` test)